pkg/kubelet/pleg/generic.go

```diff
@@ -325,29 +325,24 @@ func (g *GenericPLEG) relist() {
 func getContainersFromPods(pods ...*kubecontainer.Pod) []*kubecontainer.Container {
 	cidSet := sets.NewString()
 	var containers []*kubecontainer.Container
-	for _, p := range pods {
-		if p == nil {
-			continue
-		}
-		for _, c := range p.Containers {
-			cid := string(c.ID.ID)
-			if cidSet.Has(cid) {
-				continue
-			}
-			cidSet.Insert(cid)
-			containers = append(containers, c)
-		}
-		// Update sandboxes as containers
-		// TODO: keep track of sandboxes explicitly.
-		for _, c := range p.Sandboxes {
-			cid := string(c.ID.ID)
+	fillCidSet := func(cs []*kubecontainer.Container) {
+		for _, c := range cs {
+			cid := c.ID.ID
 			if cidSet.Has(cid) {
 				continue
 			}
 			cidSet.Insert(cid)
 			containers = append(containers, c)
 		}
+	}
 
+	for _, p := range pods {
+		if p == nil {
+			continue
+		}
+		fillCidSet(p.Containers)
+		// Update sandboxes as containers
```

Comment:
maintain the TODO for sandbox tracking please :)

Reply:
Thanks for reminding! done for that.